src/core/config/config.ts

```diff
@@ -171,27 +183,121 @@ export class Config implements IConfig {
   }
 }
 
+export function mergeConfigs(
+  originalConfig: Config,
+  newConfig: Config,
+  fileSystem: FSPath,
+  logger?: ILogger
+): Config {
+  const env = {
+    superfaceApiUrl:
+      newConfig.superfaceApiUrl ?? originalConfig.superfaceApiUrl,
+    sdkAuthToken: newConfig.sdkAuthToken ?? originalConfig.sdkAuthToken,
+    superfacePath: newConfig.superfacePath ?? originalConfig.superfacePath,
+    superfaceCacheTimeout:
+      newConfig.superfaceCacheTimeout ?? originalConfig.superfaceCacheTimeout,
+    metricDebounceTimeMin:
+      newConfig.metricDebounceTimeMin ?? originalConfig.metricDebounceTimeMin,
+    metricDebounceTimeMax:
+      newConfig.metricDebounceTimeMax ?? originalConfig.metricDebounceTimeMax,
+    disableReporting:
+      newConfig.disableReporting ?? originalConfig.disableReporting,
+    cachePath: newConfig.cachePath ?? originalConfig.cachePath,
+    sandboxTimeout: newConfig.sandboxTimeout ?? originalConfig.sandboxTimeout,
+  };
+
+  logger?.log(
+    DEBUG_NAMESPACE,
+    'Merged config A: %O with B: %O to: %O',
+    originalConfig,
+    newConfig,
+    env
+  );
+
+  return new Config(fileSystem, env);
+}
+
+export function loadConfigFromCode(
+  config: Partial<IConfig>,
+  fileSystem: FSPath,
+  logger?: ILogger
+): Config {
+  const env = {
+    superfaceApiUrl: ensureValidUrl(config.superfaceApiUrl),
+    sdkAuthToken: ensureValidSdkToken(
+      config.sdkAuthToken,
+      'sdkAuthToken',
+      logger?.log(DEBUG_NAMESPACE)
+    ),
+    // TODO: Check if it is path?
```

Comment:
:+1:

Reply:
@lukas-valenta  FS isAbsolute can be used to check if path is absolute. For relative I would probably had to use regex. I think we should leave it for now.